### ISSUE TITLE
Error when compiling latest revision of rathena

### ISSUE THREAD

**Issue by jasonch35** [rathena/rathena#6210]
* **rAthena Hash**:  6c35726

* **Client Date**: 

Not relevant

* **Server Mode**: 

Pre-Renewal

* **Description of Issue**: 
Error when compiling.
  * Result: 
  ![Screenshot 2021-08-21 221823](https://user-images.githubusercontent.com/31005928/130356840-c392190e-7c36-4e36-8a96-7e1953bf78d4.png)
  
Fresh OS install and clone of rathena. Only conf folder touched. Done 3 times with different Unix distros( Debian 10, Pop os, Debian 11. GCC version 10.

* **Modifications that may affect results**: 
  It might have something to do with the conversion to YAML where (string) wasn't include: 
  `#include <string>`

**Comment by talesvalente** [comment]
Try #include <string.h>

**Comment by mundussan** [comment]
Can't reproduce your issue.
I just installed Debian on a VM, fresh, and did the requisites installation and basic configuration of the server.
Everything worked fine.

**Comment by jasonch35** [comment]
> Can't reproduce your issue.
> I just installed Debian 10 on a VM, fresh, and did the requisites installation and basic configuration of the server.
> Everything worked fine.

Can you please try Pre-Re? Sorry I wrote renewal as server mode.

**Comment by Daegaladh** [comment]
@jasonch35 are you using GCC 9 or 10?

**Comment by mundussan** [comment]
> > Can't reproduce your issue.
> > I just installed Debian 10 on a VM, fresh, and did the requisites installation and basic configuration of the server.
> > Everything worked fine.
> 
> Can you please try Pre-Re? Sorry I wrote renewal as server mode. I uncommented #define PRERE, and disabled every renewal config on ./config/renewal.hpp.

did you tried compile using configure?

using `./configure --enable-prere` ?
or you used cmake?

**Comment by mundussan** [comment]
> > > Can't reproduce your issue.
> > > I just installed Debian 10 on a VM, fresh, and did the requisites installation and basic configuration of the server.
> > > Everything worked fine.
> > 
> > 
> > Can you please try Pre-Re? Sorry I wrote renewal as server mode. I uncommented #define PRERE, and disabled every renewal config on ./config/renewal.hpp.
> 
> did you tried compile using configure?
> 
> using `./configure --enable-prere` ?
> or you used cmake?

since for test i'm using configure and make.
and also, prere run fine too here on VM.

**Comment by Daegaladh** [comment]
@mundussan what GCC version are you using for your tests?
I've noticed people with GCC 9 or 10 getting some compiling errors lately

**Comment by mundussan** [comment]
> @mundussan what GCC version are you using for your tests?
> I've noticed people with GCC 9 or 10 getting some compiling errors lately

almost sure it is 9.3, let me check here.
its from stable repo from debian 10.

**Comment by jasonch35** [comment]
@Daegaladh @mundussan 

I inserted `#include <string>` in map.hpp, error was gone. Now it's having this vending warning, I don't know if it has something to do with the workaround.
![vending](https://user-images.githubusercontent.com/31005928/130458510-82757f1b-0258-4378-9add-fc104c7a682a.png)


But my paranoid mind tells me that my server might run into more errors because the line wasn't intended to be there.

I'm using GCC 10. I can't downgrade my GCC to 7 or 8, I've tried, errors everywhere.
